test/Microsoft.AspNetCore.Mvc.RazorPages.Test/Infrastructure/PageDirectiveFeatureTest.cs

```diff
@@ -0,0 +1,243 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.IO;
+using System.Text;
+using Microsoft.AspNetCore.Razor.Evolution;
+using Xunit;
+
+namespace Microsoft.AspNetCore.Mvc.RazorPages.Infrastructure
+{
+    public class PageDirectiveFeatureTest
```

Comment:
It's my understanding that @rynowak wasn't wanting to mess with parsing and verifying these too much since we're eventually going to transition to using razor parsing which would deal with a lot of this for us. As written if you have multiple quotes it will include the extras as part of the template and I've added a test which proves that. I think this boils down to how much verifying of the template we want to do in this function. What if we have `"this/is/"a/template"` or `"this/is/'a/template"`, both of these are invalid url parts, but is it this functions job to ensure that `out template` is a valid part of a url?